src/main/java/org/gradle/profiler/studio/StudioGradleClient.java

```diff
@@ -38,10 +48,10 @@ public StudioGradleClient(GradleBuildConfiguration buildConfiguration, Invocatio
         }
         System.out.println("* Main class: " + launchConfiguration.getMainClass());
 
-        server = new Server("agent");
-        studioProcess = startStudio(launchConfiguration, studioInstallDir, invocationSettings, server);
-        agentConnection = server.waitForIncoming(Duration.ofMinutes(1));
-        agentConnection.send(new ConnectionParameters(buildConfiguration.getGradleHome()));
+        studioProcess = startStudio(launchConfiguration, studioInstallDir, invocationSettings, studioAgentServer);
+        studioPluginConnection = studioPluginServer.waitForIncoming(Duration.ofMinutes(1));
+        studioAgentConnection = studioAgentServer.waitForIncoming(Duration.ofMinutes(1));
```

Comment:
Should we extract the `Duration.ofMinutes(1)` into some constant in this class? Do you think this needs to be parameterized?